dbt_semantic_interfaces/implementations/element_config.py

```diff
@@ -0,0 +1,18 @@
+from typing import Any, Dict
+
+from pydantic import Field
+from typing_extensions import override
+
+from dbt_semantic_interfaces.implementations.base import HashableBaseModel
+from dbt_semantic_interfaces.protocols.meta import SemanticLayerElementConfig
+from dbt_semantic_interfaces.protocols.protocol_hint import ProtocolHint
+
+
+class PydanticSemanticLayerElementConfig(HashableBaseModel, ProtocolHint[SemanticLayerElementConfig]):
+    """PydanticDimension config."""
+
+    @override
+    def _implements_protocol(self) -> SemanticLayerElementConfig:  # noqa: D
+        return self
+
+    meta: Dict[str, Any] = Field(default_factory=dict)
```

Comment:
Would this work if we add more fields to the config and wanted to treat meta as optional?  (For example, if we wanted to allow someone to add either some metadata or a tag.)

(This is not necessarily a blocking question.)

Reply:
Good point! We can make this optional in the future, though, without a breaking change.

Reply:
Is this not already optional? In a sense that, it'll default to an empty dict if you didn't provide it anything

Reply:
Yet it is effectively optional because of the default, although not type `Optional`.